server/src/main/java/org/tctalent/server/service/db/impl/CandidateServiceImpl.java

```diff
@@ -405,2392 +405,2410 @@
+      if (taskAssignment instanceof QuestionTaskAssignmentImpl) {
+        QuestionTaskAssignment qta = (QuestionTaskAssignmentImpl) taskAssignment;
+        String answer = fetchCandidateTaskAnswer(qta);
+        qta.setAnswer(answer);
+      }
+    }
 
-            log.info("Found " + candidates.getTotalElements() + " candidates in search");
-            return candidates;
+  }
+
+  /**
+   * Retrieves the answer, if any, of the give question task assignment.
+   *
+   * @param questionTaskAssignment Question task assignment
+   * @return Candidate's answer to the question
+   * @throws NoSuchObjectException if the answer could not be retrieved because the answer has been specified as being located in a non
+   *                               existent candidate field or property.
+   * @see #storeCandidateTaskAnswer
+   */
+  @Nullable
+  private String fetchCandidateTaskAnswer(QuestionTaskAssignment questionTaskAssignment)
+      throws NoSuchObjectException {
+    String answer;
+    Task task = questionTaskAssignment.getTask();
+    if (task instanceof QuestionTask) {
+      String answerField = ((QuestionTask) task).getCandidateAnswerField();
+      Candidate candidate = questionTaskAssignment.getCandidate();
+      if (answerField == null) {
+        //Get answer from candidate property
+        String propertyName = task.getName();
+        final CandidateProperty property =
+            candidatePropertyService.findProperty(candidate, propertyName);
+        answer = property != null ? property.getValue() : null;
+      } else {
+        //Get answer from candidate field
+
+        //TODO JC This is a bit of a hack - we need a better way of processing things like candidateExams
+        String candidateExamsFieldName = "candidateExams";
+        if (answerField.startsWith(candidateExamsFieldName + ".")) {
+          //Special code for candidate exams
+          //Extract the exam name from the answerField
+          String examName = answerField.substring(candidateExamsFieldName.length() + 1);
+          Exam examType = Exam.valueOf(examName);
+
+          //See if we already have an entry for this exam
+          Optional<CandidateExam> examO = Optional.empty();
+          final List<CandidateExam> candidateExams = candidate.getCandidateExams();
+          if (candidateExams != null) {
+            examO = candidateExams.stream().filter(
+                e -> e.getExam().equals(examType)).findFirst();
+          }
+
+          //Fetch the answer
+          CandidateExam exam;
+          if (examO.isPresent()) {
+            exam = examO.get();
+            answer = exam.getScore();
+          } else {
+            answer = null;
+          }
         } else {
-            return null;
+          try {
+            Object value = PropertyUtils.getProperty(candidate, answerField);
+            answer = value != null ? value.toString() : null;
+          } catch (IllegalAccessException e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While re-throwing the caught exception with modifications , information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)
Similar issue at line numbers 585, 588.